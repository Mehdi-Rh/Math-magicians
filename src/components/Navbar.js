import React, { useState } from 'react';
import {
  Link,
} from 'react-router-dom';
import './Navbar.css';

const Navbar = () => {
  const [isNavExpanded, setIsNavExpanded] = useState(false);

  return (
    <nav className="navigation">
      <a href="/" className="brand-name">
        Math Magicians
      </a>
      <button
        type="button"
        className="hamburger"
        onClick={() => {
          setIsNavExpanded(!isNavExpanded);
        }}
      >
        {/* icon from heroicons.com */}
        <svg
          xmlns="http://www.w3.org/2000/svg"
          className="h-5 w-5 "
          viewBox="0 0 20 20"
          fill="#86c232"
        >
          <path
            fillRule="evenodd"
            d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM9 15a1 1 0 011-1h6a1 1 0 110 2h-6a1 1 0 01-1-1z"
            clipRule="evenodd"
            className="HamburgerLines"
          />
        </svg>
      </button>
      <div
        className={
          isNavExpanded ? 'navigation-menu expanded' : 'navigation-menu'
        }
      >
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/calculator">Calculator</Link>
          </li>
          <li>
            <Link to="/quote">Quote</Link>
          </li>
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
