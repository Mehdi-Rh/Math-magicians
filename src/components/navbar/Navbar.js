import React, { useEffect, useState } from 'react';
import {
  Link,
} from 'react-router-dom';
import './Navbar.css';
import { fadeInRight, fadeInLeft } from 'react-animations';
import Radium, { StyleRoot } from 'radium';

const styles = {
  fadeInRight: {
    animation: 'x 2s',
    animationName: Radium.keyframes(fadeInRight, 'fadeInRight'),
  },
  fadeInLeft: {
    animation: 'x 2s',
    animationName: Radium.keyframes(fadeInLeft, 'fadeInLeft'),
  },

};

const Navbar = () => {
  const [isNavExpanded, setIsNavExpanded] = useState(false);

  useEffect(() => {
    // if (window.innerWidth > 768) {
    //   setIsNavExpanded(false);
    //   console.log('resize');
    // }
    const handleResize = () => {
      if (window.innerWidth > 768) {
        setIsNavExpanded(false);
      }
      console.log(isNavExpanded);
      console.log('resized to: ', window.innerWidth, 'x', window.innerHeight);
    };

    window.addEventListener('resize', handleResize);
  });

  return (
    <StyleRoot>
      <header className="navigation">
        <a href="/" style={(window.innerWidth > 768) ? styles.fadeInLeft : {}} className="brand-name">
          Math Magicians
        </a>
        <button
          type="button"
          className="hamburger"
          onClick={() => {
            setIsNavExpanded(!isNavExpanded);
          }}
        >
          {/* icon from heroicons.com */}
          <svg
            xmlns="http://www.w3.org/2000/svg"
            className="h-5 w-5 "
            viewBox="0 0 20 20"
            fill="#86c232"
          >
            <path
              fillRule="evenodd"
              d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM9 15a1 1 0 011-1h6a1 1 0 110 2h-6a1 1 0 01-1-1z"
              clipRule="evenodd"
              className="HamburgerLines"
            />
          </svg>
        </button>
        <div
          className={
          isNavExpanded ? 'navigation-menu expanded' : 'navigation-menu'
        }
        >
          <ul style={(window.innerWidth > 768) ? styles.fadeInRight : {}} id="nav">
            <li>
              <Link
                onClick={() => {
                  setIsNavExpanded(!isNavExpanded);
                }}
                to="/"
              >
                Home
              </Link>
            </li>
            <li>
              <Link
                onClick={() => {
                  setIsNavExpanded(!isNavExpanded);
                }}
                to="/calculator"
              >
                Calculator
              </Link>
            </li>
            <li>
              <Link
                onClick={() => {
                  setIsNavExpanded(!isNavExpanded);
                }}
                to="/quote"
              >
                Quote
              </Link>
            </li>
          </ul>
        </div>
      </header>
    </StyleRoot>

  );
};

export default Navbar;
