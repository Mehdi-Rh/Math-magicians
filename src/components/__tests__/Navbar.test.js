import React from 'react';
import Renderer from 'react-test-renderer';
import Navbar from '../Navbar';

it('renders correctly', () => {
  const tree = Renderer
    .create(<Navbar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
