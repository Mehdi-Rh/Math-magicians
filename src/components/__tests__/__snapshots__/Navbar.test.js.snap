// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="navigation"
>
  <a
    className="brand-name"
    href="/"
  >
    Math Magicians
  </a>
  <button
    className="hamburger"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="h-5 w-5 "
      fill="#86c232"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="HamburgerLines"
        clipRule="evenodd"
        d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM9 15a1 1 0 011-1h6a1 1 0 110 2h-6a1 1 0 01-1-1z"
        fillRule="evenodd"
      />
    </svg>
  </button>
  <div
    className="navigation-menu"
  >
    <ul>
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/calculator"
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          href="/quote"
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
